buildscripts/generate-version-file.gradle

```diff
@@ -16,9 +16,9 @@
  */
 
 def createVersionFile = tasks.register('createVersionFile') {
-    onlyIf {
-        project.hasProperty("lastCommitHash")
-    }
+    def commitHash = project.hasProperty("lastCommitHash")
+            ? project.property("lastCommitHash")
+            : 'git rev-parse HEAD'.execute().text.trim()
```

Comment:
Let's pass the root directory as a working directory and check the result of the execution